### ISSUE TITLE
Documentation for flattenRing doesn't fully describe functionality

### ISSUE THREAD

**Issue by kschwede** [Macaulay2/M2#651]
Sorry if this isn't the right place (ie, do Documentation Issues go here?), but the function flattenRing asserts:

> write a ring as a (quotient) of a polynomial ring over ZZ or a prime field

However, as far I can tell, it also works for things over QQ, Galois fields, function fields, etc.  The lower part of the documentation even seems to acknowledge this (ie, you can use toField...)

**Comment by lilysilverstein** [comment]
Indeed, several of the methods for flattenRing are currently undocumented. Will be fixed shortly.

**Comment by lilysilverstein** [comment]
Rewrote this documentation to include things like Galois fields and rings declared as fields.

I was extremely confused by the description of the Result option. I changed the page to go into much more detail, with more examples, of the various numbers, types, and sequences that can all be passed to Result. Finally, I added documentation nodes for the options.

Commit is [here](https://github.com/lilysilverstein/M2/commit/39cc70ff91e2e24626e66a1cf7055dc86c242303), and will be bundled into my next documentation pull request.

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

Closing as fixed — this was done in 2018, reported here at the time, and just never closed.

@lilysilverstein's second comment on this thread,

> Rewrote this documentation to include things like Galois fields and rings declared as
> fields.

is `39cc70ff91` ("rewrite flattenRing and its options", 2018-07-22), carried by **#870**,
merged 2018-11-14.

### The specific complaint is gone

The headline that prompted the report is no longer there. `flattenRing-doc.m2:15` now reads

> write a ring as a (quotient of a) polynomial ring

with the "over ZZ or a prime field" clause dropped — which is the whole of what was
inaccurate.

All three cases named in the report are now documented with worked examples:

| named in the report | on the page today |
| --- | --- |
| Galois fields | *"`flattenRing` can also be used to represent GaloisFields as quotients of polynomial rings"*, with `K = GF(5^3)` |
| function fields | `K = frac(ZZ[a])`, then `B = K[x,y,z]/(...)`, showing the coefficient ring of the result is `K` |
| `toField` rings | *"Once a ring has been declared to be a field with `toField` then it will be used as the coefficient ring"*, with `A = QQ[a]/(a^2-3)` |

The page also now states the general rule the old headline was a special case of — *"By
default `flattenRing` attempts to define a new ring over either `ZZ` or a base field"* — and
the `Result` option, which @lilysilverstein said in the same comment she found confusing, was
rewritten with examples of each accepted form.

Thanks @kschwede for the report and @lilysilverstein for doing the work; the only thing left
undone was closing the issue.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>
